src/istgt_iscsi.c

```diff
@@ -1897,6 +1899,8 @@ istgt_iscsi_op_login(CONN_Ptr conn, ISCSI_PDU_Ptr pdu)
 				StatusDetail = 0x07;
 				goto response;
 			}
+			discovery_counter = 0;
```

Comment:
When 5 concurrent connections happen, this will crash even though discovery might succeed. The discovery value should be per connection and not global.

Reply:
Every retry in discovery happens on a new connection. And, we want to track this across connections. So, it need to be global.
It can be LUN specific. But, as istgt supports only one LUN at a time, global is still fine.
I will add a comment that this need to be taken care when multiple LUN support is there in istgt.

Reply:
I'm not sure I understand this but there is nothing that prevents -- multiple logins from different hosts in istgt right now. Keeping the counter global will trigger the if below, even though it's not what we wanted. 

Multiple nodes might issue a discovery command to this instance and if that number happens to be the whatever you set it to, it will abort() if you feel that that is not a problem, then OK. The point here is that it will cause the program to abort scenarios we do not necessarily want it to abort.

Reply:
multiple logins from different nodes will not happen at same time in k8s environment. So, it should be fine.